src/app/tasks/task-details/task-details.component.ts

```diff
@@ -49,30 +43,42 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
   onSubmit () {
     // Get filled out form data using form group
 
-    this.taskService.updateTask(
-      this.task.id,
-      {
-        id: this.task.id,
-        title: this.editTaskForm.get('title')?.value,
-        description: this.editTaskForm.get('description')?.value,
-        type: this.taskTypes.find( taskType => 
-          taskType.value == this.editTaskForm.get('type')?.value
-        ) as Option,
-        status: this.taskStatuses.find(taskStatus => 
-          taskStatus.value == this.editTaskForm.get('status')?.value
-        ) as Option,
-        modifiedOn: new Date(),
-        createdOn: this.task.createdOn,
-      }
-    );
-
-    this.router.navigate(['tasks']);
+    if(this.task) {
```

Comment:
to avoid this if statement you could pass task to onSubmit method as a variable in the template, since you do this check on the template level already, no need to make it twice